server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -564,19 +564,15 @@ public int decrementResourceTrackingThreads() {
         int count = numActiveResourceTrackingThreads.decrementAndGet();
 
         if (count == 0) {
-            List<Exception> listenerExceptions = new ArrayList<>();
-            resourceTrackingCompletionListeners.forEach(listener -> {
```

Comment:
@fudongyingluck  Seems like there is a potential behavior change here. The NotifyOnceListener will only take action on the first invocation, and all subsequent invocations are no-ops. You're removing that behavior by using a basic Consumer. How do you this won't cause a regression?

Reply:
Thanks for your time. I think the "all subsequent invocations" means the innerOnFailure(). Although there is a little change here, I do not think we should surround the exception layer after layer. The code is very hard to read. And innerOnFailure() actually has another way to complete it. Since I haven't worked on this part recently,  this issue will be closed. Thanks for your time again ~